target/linux/ramips/patches-4.14/0039-mtd-add-mt7621-nand-support.patch

```diff
@@ -3297,13 +3297,13 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
 +			printk("compare signature failed %x\n", page);
 +			return -1;
 +		}
-+		if (mtk_nand_exec_read_page(mtd, page, mtd->writesize, chip->buffers->databuf, chip->oob_poi))
++		if (mtk_nand_exec_read_page(mtd, chip->pagebuf = page, mtd->writesize, chip->buffers->databuf, chip->oob_poi))
```

Comment:
This looks weird.

Reply:
If you are talking about the assign-and-use, I have refactored it now to set pagebuf after a successful read. If this is not what you refer to as weird, please clarify. Thanks.